crates/relayer/relayer-components/src/relay/impls/packet_clearers/ack.rs

```diff
@@ -0,0 +1,99 @@
+use cgp_core::async_trait;
+
+use crate::chain::traits::queries::packet_acknowledgements::CanQueryPacketAcknowledgements;
+use crate::chain::traits::queries::packet_commitments::CanQueryPacketCommitments;
+use crate::chain::traits::queries::send_packets::CanQuerySendPackets;
+use crate::chain::traits::queries::unreceived_acks_sequences::CanQueryUnreceivedAcksSequences;
+use crate::chain::types::aliases::{ChannelIdOf, PortIdOf};
+use crate::relay::traits::chains::{CanRaiseRelayChainErrors, HasRelayChains};
+use crate::relay::traits::packet_clearer::PacketClearer;
+use crate::relay::traits::packet_relayer::CanRelayPacket;
+use crate::runtime::traits::runtime::HasRuntime;
+use crate::runtime::traits::task::{CanRunConcurrentTasks, Task};
+
+pub struct ClearAckPackets;
+
+pub struct RelayPacketTask<Relay>
+where
+    Relay: HasRelayChains,
+{
+    pub relay: Relay,
+    pub packet: Relay::Packet,
+}
+
+#[async_trait]
+impl<Relay> Task for RelayPacketTask<Relay>
+where
+    Relay: CanRelayPacket,
+{
+    async fn run(self) {
+        let _ = self.relay.relay_packet(&self.packet).await;
```

Comment:
We should probably deal with the error that is potentially returned here, eg. log it.

Reply:
Do you mean something like this?
```suggestion
        let result = self.relay.relay_packet(&self.packet).await;
        if let Err(e) = result {
            let error_desc = format!("{e:#?}");
            self.relay.log_error(&error_desc);
        }
```

Reply:
Yep! Perhaps with a message indicating we failed to relay.

Reply:
I think we would need to add the constraint `HasRelayPacketFields` which might not be the case for non-Cosmos packets.
cc @soareschen

Reply:
Those fields are defined in the [Packet interface in ICS04](https://github.com/cosmos/ibc/blob/main/spec/core/ics-004-channel-and-packet-semantics/README.md) so all IBC-enabled chains should have them IIUC, no?

Reply:
I think you only need `Relay::Packet: Debug` or `Display` for formatting the packet.

Reply:
done in faa05f1536a7af583e424b55c81f00a24510cfe1